### PR TITLE
Added Parameter Object With Undefined Type query for OpenAPI #2846

### LINKED ISSUE #2846: Add Parameter Object With Undefined Type query for OpenAPI
### Platform
OpenAPI 3.0

### Description
A Parameter Object must contain either a 'schema' property, or a 'content' property

### PR DESCRIPTION
**Pull request by joaoReigota1**
Signed-off-by: João Reigota <joao.reigota@checkmarx.com>

Closes #2846

**Proposed Changes**
- Added Parameter Object With Undefined Type query for OpenAPI

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/parameter_object_undefined_type/query.rego [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+	params := doc.paths[name].parameters[n]
```
Comment (rafaela-soares) [inline assets/queries/openAPI/parameter_object_undefined_type/query.rego:8]: Parameters can be defined in other attributes - https://swagger.io/docs/specification/describing-parameters/#path-parameters. For example, path[name].get.parameters

File: assets/queries/openAPI/parameter_object_undefined_type/metadata.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "46facedc-f243-4108-ab33-583b807d50b0",
+  "queryName": "Parameter Object With Undefined Type",
```
Comment (rafaela-soares) [inline assets/queries/openAPI/parameter_object_undefined_type/metadata.json:3]: The folder name is slightly different from the 'queryName' in the metadata file

Comment (rogeriopeixotocx) [reply assets/queries/openAPI/parameter_object_undefined_type/metadata.json:3]: I think it's ok

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**